src/types/user.ts

```diff
@@ -90,20 +92,32 @@ export function validateSpidUser(value: any): Either<Error, SpidUser> {
     FISCAL_NUMBER_INTERNATIONAL_PREFIX,
     ""
   );
-  const valueWithoutPrefix = {
-    ...value,
-    fiscalNumber: fiscalNumberWithoutPrefix
-  };
+
+  const maybeAuthnContextClassRef = getAuthnContextFromResponse(
+    value.getAssertionXml()
+  );
 
   // Set SPID level to a default (SPID_L2) if the expected value is not available
   // in the SAML assertion.
   // Actually the value returned by the test idp is invalid
   // @see https://github.com/italia/spid-testenv/issues/26
+  const authnContextClassRef = maybeAuthnContextClassRef
+    .filter(isSpidL)
+    .getOrElse(SpidLevelEnum["https://www.spid.gov.it/SpidL2"]);
+
+  log.info(
+    "Response from IDP (authnContextClassRef): %s",
+    authnContextClassRef
+  );
+
+  const valueWithoutPrefix = {
+    ...value,
+    fiscalNumber: fiscalNumberWithoutPrefix
+  };
+
   const valueWithDefaultSPIDLevel = {
     ...valueWithoutPrefix,
-    authnContextClassRef: isSpidL(valueWithoutPrefix.authnContextClassRef)
```

Comment:
valueWithoutPrefix.authnContextClassRef is always undefined, we must extract the value from the SPID response